File: docs/snippets/all/archetypes/mcap_channel_simple.cpp
// Log a simple MCAP channel definition.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_mcap_channel");
    rec.spawn().exit_on_failure();

    const std::vector<rerun::datatypes::Utf8Pair> metadata = {
        {"frame_id", "camera_link"},
        {"encoding", "bgr8"},
    };

    rec.log(
        "mcap/channels/camera",
        rerun::archetypes::McapChannel(1, "/camera/image", "cdr")
            .with_metadata(rerun::components::KeyValuePairs(metadata))
    );
}

File: docs/snippets/all/archetypes/mcap_channel_simple.py
"""Log a simple MCAP channel definition."""

import rerun as rr

rr.init("rerun_example_mcap_channel", spawn=True)

rr.log(
    "mcap/channels/camera",
    rr.McapChannel(
        id=1,
        topic="/camera/image",
        message_encoding="cdr",
        metadata={"frame_id": "camera_link", "encoding": "bgr8"},
    ),
)
